scripts/quests/sandoria/A_Knights_Test.lua

```diff
@@ -0,0 +1,169 @@
+-----------------------------------
+--  A Knights Test
+--  NPC: Balasiel
+--  !pos -136 -11 64 230
+--  !addquest 0 29
+--  Cahaurme !pos 55.749 -8.601 -29.354 230
+--  Baunise !pos -55 -8 -32 230
+--  Disused Well !pos -221 2 -293 149
+-----------------------------------
+require('scripts/globals/items')
+require("scripts/globals/keyitems")
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require("scripts/globals/status")
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local ID = require("scripts/zones/Southern_San_dOria/IDs")
+local DavoiID = require("scripts/zones/Davoi/IDs")
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.A_KNIGHT_S_TEST)
+
+quest.reward =
+{
+    fame     = 30,
+    fameArea = xi.quest.fame_area.SANDORIA,
+    item     = xi.items.KITE_SHIELD,
+    title    = xi.title.TRIED_AND_TESTED_KNIGHT,
+
+}
+
+quest.sections =
+{
+    -- Section: Quest available.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getQuestStatus(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.A_SQUIRE_S_TEST_II) == QUEST_COMPLETED and
+                player:getMainLvl() >= 30
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Balasiel'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(627)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [627] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                        player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.BOOK_OF_TASKS)
+                        player:addKeyItem(xi.ki.BOOK_OF_TASKS)
+                    else
+                        return
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Balasiel'] =
+            {
+                onTrigger = function(player, npc)
+                    if not player:hasKeyItem (xi.ki.KNIGHTS_SOUL) then
+                        return quest:progressEvent (669)
+                    elseif player:hasKeyItem (xi.ki.KNIGHTS_SOUL) then
+                        return quest:progressEvent (628)
+                    else
+                        return quest:startEvent (667)
```

Comment:
this else will never be reached. The two above if statements will always be checked instead.